### PR TITLE
[GQL-2] #4972 Event triggers Advanced settings - Unselect all columns or start with no selected columns

### PR DESCRIPTION
**Pull request by sauzzon**
### Description

When creating Event Triggers in Hasura, in the Advanced Settings you can select which columns the trigger should be run on. The UI starts with all columns selected but it should be possible to unselect all columns if you have many columns and only want to select a few. This PR adds two buttons (Unselect all and select all) so that it becomes easy for the users to select the columns they want.

### Affected components

- [x] Console

### Related Issues

#4972

### Solution and Design

This issue is fixed by creating two buttons in the UI. Select all and Unselect all buttons are created so that the selection of the columns becomes easy and the raised issue gets solved.

https://user-images.githubusercontent.com/59878660/174267137-c22ea3a5-58ba-40ac-8df1-87102baf556b.mp4

### Steps to test and verify

1. Go to Advanced Settings while creating event triggers in hasura.
2. Click on the unselect all button to unselect all the columns
3. Choose the columns you want to select
4. Also try to select all by clicking Select all button

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [x] No

#### Metadata

Does this PR add a new Metadata feature?
- [x] No

#### GraphQL
- [x] No new GraphQL schema is generated

#### Breaking changes

- [x] No Breaking changes

### CODE REVIEW

File: server/src-lib/Hasura/GraphQL/Parser/Class.hs [hunk 1]
```diff
@@ -93,7 +91,7 @@ class (Monad m, MonadParse n) => MonadSchema n m | m -> n where
   -- @
   memoizeOn ::
     forall p a b.
-    (HasCallStack, Ord a, Typeable p, Typeable a, Typeable b) =>
```
Comment (bipsBro) [inline server/src-lib/Hasura/GraphQL/Parser/Class.hs:96]: Why did you remove this @sauzzon?

File: server/src-lib/Hasura/GraphQL/Parser/Class.hs [hunk 2]
```diff
@@ -108,8 +106,8 @@ class (Monad m, MonadParse n) => MonadSchema n m | m -> n where
...
 -- as the key.
 memoize ::
-  (HasCallStack, MonadSchema n m, Ord a, Typeable a, Typeable b, Typeable k) =>
+  (MonadSchema n m, Ord a, Typeable a, Typeable b, Typeable p) =>
   TH.Name ->
-  (a -> m (Parser k n b)) ->
-  (a -> m (Parser k n b))
+  (a -> m (p n b)) ->
+  (a -> m (p n b))
 memoize name f a = memoizeOn name a (f a)
```
Comment (bipsBro) [inline server/src-lib/Hasura/GraphQL/Parser/Class.hs:113]: Are these necessary changes?

File: server/src-lib/Hasura/GraphQL/Parser/Class.hs [hunk 3]
```diff
@@ -7,9 +7,7 @@ module Hasura.GraphQL.Parser.Class
   )
 where
 
-import GHC.Stack (HasCallStack)
 import Hasura.GraphQL.Parser.Class.Parse
-import Hasura.GraphQL.Parser.Internal.Types
```
Comment (bipsBro) [inline server/src-lib/Hasura/GraphQL/Parser/Class.hs:12]: any reason to remove this @sauzzon ?

**Review by bipsBro — CHANGES_REQUESTED** [review]
Need some clarification.

File: server/src-lib/Hasura/GraphQL/Parser/Class.hs (hunk 1 above)
Comment (sauzzon) [reply server/src-lib/Hasura/GraphQL/Parser/Class.hs]: it wasn't done by me. Sync issues

File: server/src-lib/Hasura/GraphQL/Parser/Class.hs (hunk 3 above)
Comment (sauzzon) [reply server/src-lib/Hasura/GraphQL/Parser/Class.hs]: it wasn't done by me. Sync issues

File: server/src-lib/Hasura/GraphQL/Parser/Class.hs (hunk 2 above)
Comment (sauzzon) [reply server/src-lib/Hasura/GraphQL/Parser/Class.hs]: it wasn't done by me. Sync issues

**Review by sudeshna-22 — APPROVED** [review]

**Review by shrestharajani — APPROVED** [review]

**Pull request closed without merging**